engine/src/main/java/org/destinationsol/game/screens/MapScreen.java

```diff
@@ -76,5 +101,94 @@ public void updateCustom(SolApplication solApplication, SolInputManager.InputPoi
                 zoomInControl.maybeFlashPressed(gameOptions.getKeyZoomIn());
             }
         }
+
+        if (addWaypointControl.isJustOff() && !isPickingWaypointSpot) {
+            isPickingWaypointSpot = true;
+            addWaypointControl.setDisplayName(CANCEL_TEXT);
+            removeWaypointControl.setEnabled(false);
+        } else if (addWaypointControl.isJustOff() && isPickingWaypointSpot) {
+            isPickingWaypointSpot = false;
+            addWaypointControl.setDisplayName(NEW_WAYPOINT_TEXT);
+            removeWaypointControl.setEnabled(true);
+        } else if (isPickingWaypointSpot) {
+            if (inputPointers[0].isJustUnPressed()) {
+                Vector2 clickPosition = new Vector2(inputPointers[0].x, inputPointers[0].y);
+
+                Vector2 worldPosition = screenPositionToWorld(clickPosition, game.getCam().getPosition(), mapZoom);
+
+                ArrayList<Waypoint> waypoints = game.getHero().getWaypoints();
+                boolean canCreate = true;
+                for (int w = 0; w < waypoints.size(); w++) {
+                    Waypoint waypoint = waypoints.get(w);
+                    if (worldPosition.x > waypoint.position.x-MIN_WAYPOINT_DIST && worldPosition.x < waypoint.position.x+MIN_WAYPOINT_DIST &&
+                        worldPosition.y > waypoint.position.y-MIN_WAYPOINT_DIST && worldPosition.y < waypoint.position.y+MIN_WAYPOINT_DIST) {
+                        canCreate = false;
+                        break;
+                    }
+                }
+
+                if (canCreate) {
+                    disableWaypointButtons(true);
+                    WaypointCreationScreen waypointCreationScreen = game.getScreens().waypointCreationScreen;
+                    waypointCreationScreen.setWaypointPos(worldPosition);
+
+                    solApplication.getInputManager().setScreen(solApplication, game.getScreens().mapScreen);
+                    solApplication.getInputManager().addScreen(solApplication, waypointCreationScreen);
+                }
+                addWaypointControl.setDisplayName(NEW_WAYPOINT_TEXT);
+                isPickingWaypointSpot = false;
+            }
+        }
+
+        if (removeWaypointControl.isJustOff() && !isPickingWaypointToRemove) {
+            isPickingWaypointToRemove = true;
+            removeWaypointControl.setDisplayName(CANCEL_TEXT);
+            addWaypointControl.setEnabled(false);
+        } else if (removeWaypointControl.isJustOff() && isPickingWaypointToRemove) {
+            removeWaypointControl.setDisplayName(REMOVE_WAYPOINT_TEXT);
+            addWaypointControl.setEnabled(true);
+            isPickingWaypointToRemove = false;
+        }
+        else if (isPickingWaypointToRemove) {
```

Comment:
The formatting seems a bit inconsistent here. Shouldn't the elseif be one line up?